test/Dialect/XTenNN/quantize.mlir

```diff
@@ -3,65 +3,133 @@
     return %result : tensor<1x3xi8>
 }
 
 // -----
 
 func.func @invalid_input_type(%arg0: tensor<1x2xi32>) -> tensor<1x2xi8> {
-    // expected-error@+1 {{op operand #0 must be tensor of 32-bit float values, but got 'tensor<1x2xi32>'}}
-    %result = xten_nn.quantize (%arg0: tensor<1x2xi32>) {shift = -1: si32} -> tensor<1x2xi8>
+    // expected-error@+1 {{op operand #0 must be tensor of floating-point values, but got 'tensor<1x2xi32>'}}
+    %result = xten_nn.quantize (%arg0: tensor<1x2xi32>) {shift = -1: si32, scale = 0.5 : f32, zero_point = 0 : i8} -> tensor<1x2xi8>
     return %result : tensor<1x2xi8>
 }
 // -----
 
 func.func @different_bitwidth(%arg0: tensor<1x2xf32>) -> tensor<1x2xi3> {
-    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {shift = -1: si32} -> tensor<1x2xi3>
+    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {shift = -1: si32, scale = 0.5 : f32, zero_point = 0 : i3} -> tensor<1x2xi3>
     return %result : tensor<1x2xi3>
 }
 
 // -----
 
 func.func @sixteen_bitwidth(%arg0: tensor<1x2xf32>) -> tensor<1x2xi16> {
-    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {shift = -1: si32} -> tensor<1x2xi16>
+    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {shift = -1: si32, scale = 0.5 : f32, zero_point = 0 : i16} -> tensor<1x2xi16>
     return %result : tensor<1x2xi16>
+}
+
+// -----
+
+func.func @valid_quantize_no_shift(%arg0: tensor<1x2xf32>) -> tensor<1x2xi8> {
+    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {scale = 0.125 : f32, zero_point = 3 : i8} -> tensor<1x2xi8>
+    return %result : tensor<1x2xi8>
+}
+
+// -----
+
+func.func @valid_quantize_no_zero_point(%arg0: tensor<1x2xf32>) -> tensor<1x2xi8> {
+    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {scale = 0.125 : f32} -> tensor<1x2xi8>
+    return %result : tensor<1x2xi8>
+}
+
+// -----
+
+func.func @valid_quantize_only_shift(%arg0: tensor<1x2xf32>) -> tensor<1x2xi8> {
+    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {shift = -2 : si32} -> tensor<1x2xi8>
+    return %result : tensor<1x2xi8>
+}
+
+// -----
+
+func.func @invalid_quantize_shift_and_zero(%arg0: tensor<1x2xf32>) -> tensor<1x2xi8> {
+    // expected-error@+1 {{It is only allowed to set a zero point if scale is set too}}
+    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {shift = -2 : si32, zero_point = 3 : i8} -> tensor<1x2xi8>
+    return %result : tensor<1x2xi8>
+}
+
+// -----
+
+func.func @invalid_no_shift_or_scale(%arg0: tensor<1x2xf32>) -> tensor<1x2xi8> {
+    // expected-error@+1 {{Shift and scale are both missing}}
+    %result = xten_nn.quantize (%arg0: tensor<1x2xf32>) {} -> tensor<1x2xi8>
+    return %result : tensor<1x2xi8>
+}
+
+// -----
+
+func.func @valid_quantize_op_generic(%arg0: tensor<1x2xf32>) -> tensor<1x2xi8> {
+    %result = "xten_nn.quantize" (%arg0) {shift = -3: si32, scale = 0.125 : f32, zero_point = 0 : i8} : (tensor<1x2xf32>) -> tensor<1x2xi8>
```

Comment:
Note: In the generic form the custom parser gets skipped, so all attributes need to be set. This is verifed by mlir itself